cmd/node-termination-handler.go

```diff
@@ -115,15 +137,22 @@ func watchForCancellationEvents(cancelChan <-chan drainevent.DrainEvent, drainEv
 		log.Printf("Got cancel event from channel %+v\n", drainEvent)
 		drainEventStore.CancelDrainEvent(drainEvent.EventID)
 		if drainEventStore.ShouldUncordonNode() {
-			node.Uncordon()
+			log.Println("Uncordoning the node due to a cancellation event")
+			err := node.Uncordon()
+			if err != nil {
+				log.Printf("Uncordoning the node failed: %v", err)
+			}
+			node.RemoveNTHLabels()
+		} else {
+			log.Println("Another drain event is active, not uncordoning the node")
 		}
 	}
 }
 
-func drainIfNecessary(drainEventStore *draineventstore.Store, node *node.Node, nthConfig config.Config) {
+func drainIfNecessary(drainEventStore *draineventstore.Store, node node.Node, nthConfig config.Config) {
 	if drainEvent, ok := drainEventStore.GetActiveEvent(); ok {
 		if drainEvent.PreDrainTask != nil {
-			err := drainEvent.PreDrainTask(node)
+			err := drainEvent.PreDrainTask(*drainEvent, node)
```

Comment:
I was trying to consistently pass with copy-by-value as opposed to pointers since, from what I was reading, it's not terribly expensive to do so (NTH doesn't need to be blazing fast either). `Node` is never modified, so the copy-by-value codifies the immutability a bit (although I realize the pointers in the node struct are copied so they are still mutable). I think this makes more sense when passing `config` around since it should never be modified and this protects it a bit. I was just trying to be consistent with `node`, although looks like I missed a spot. Do you have strong opinions on the copy-by-value approach?

Reply:
No super-strong opinions. I was just curious why the change. :)

My preference is for consistency either way.